modules/user_accounts/jsx/permissions.js

```diff
@@ -0,0 +1,341 @@
+      } else {
+        permissionUpdate[permissionIndex].checked = true;
+      }
+    }
+
+    // Update roles in case where additional roles were granted with the new permissions
+    roleUpdate = this.calculateRoles(roleUpdate, permissionUpdate);
+
+    this.setState({
+      roles: roleUpdate,
+      permissions: permissionUpdate
+    });
+  },
+  handleRoleMouseOver: function(roleID) {
+    var index = this.getRoleIndex(roleID);
+    var permissionsToHighlight = this.state.roles[index].permissions;
+    var permissionsUpdate = this.state.permissions;
+
+    for (var i = 0; i < permissionsToHighlight.length; i++) {
+      var permissionIndex = this.getPermissionIndex(permissionsToHighlight[i].permissionID);
+      permissionsUpdate[permissionIndex].highlight = true;
+    }
+
+    this.setState({
+      permissions: permissionsUpdate
+    });
+  },
+  handleRoleMouseOut: function() {
+    var permissionsUpdate = this.state.permissions;
+
+    for (var i = 0; i < permissionsUpdate.length; i++) {
+      permissionsUpdate[i].highlight = false;
+    }
+
+    this.setState({
+      permissions: permissionsUpdate
+    });
+  },
+  handlePermissionChange: function(permissionID) {
+    // Set permission state
+    var index = this.getPermissionIndex(permissionID);
+    var permissionUpdate = this.state.permissions;
+    var currentState = permissionUpdate[index].checked;
+    permissionUpdate[index].checked = !currentState;
+
+    // Update roles if needed
+    if (this.state.rolesUsed) {
+      var roleUpdate = this.calculateRoles(this.state.roles, permissionUpdate);
+      this.setState({
+        roles: roleUpdate,
+        permissions: permissionUpdate
+      });
+    } else {
+      this.setState({
+        permissions: permissionUpdate
+      });
+    }
+  },
+  render: function() {
+    if (this.state.rolesUsed) {
+      return (
+        <div className="Access">
+          <label className="col-sm-2">Access</label>
+          <div className="col-sm-2">
+            <label>Roles</label>
+            <RoleList
+            data={this.state.roles}
+            onChange={this.handleRoleChange}
+            onMouseOver={this.handleRoleMouseOver}
+            onMouseOut={this.handleRoleMouseOut} />
+          </div>
+          <div className="col-sm-8">
+            <label>Permissions</label>
+            <PermissionList
+            data={this.state.permissions}
+            onChange={this.handlePermissionChange} />
+          </div>
+        </div>
+      );
+    }
+    return (
+      <div className="Access">
+        <label className="col-sm-2">Access</label>
+        <div className="col-sm-10">
+          <label>Permissions</label>
+          <PermissionList
+          data={this.state.permissions}
+          onChange={this.handlePermissionChange} />
+        </div>
+      </div>
+    );
+  }
+});
+
+window.onload = function() {
+  var access = <Access
+    dataURL={`${loris.BaseURL}/user_accounts/ajax/getData.php`}
```

Comment:
@taracampbell Failed to load any data

EDIT: replace with `getData` with `getRolePermissionData`

![screenshot from 2017-04-06 15-32-45](https://cloud.githubusercontent.com/assets/15268287/24772207/65ce420c-1ade-11e7-91be-7c7a620717fa.png)